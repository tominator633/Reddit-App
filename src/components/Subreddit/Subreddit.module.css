* {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    font-family: "Urbanist", sans-serif;
}

/* .gb {
    border: 1px solid red;
}    */


.srCon {
    width: 150px;
    height: 150px;
    border-radius: 20px;
    display: flex;
    flex-direction: column;
    background-color: white;
    box-shadow: rgba(149, 157, 165, 0.2) 0px 8px 24px;
}

.srBanner {
    width: 100%;
    height: 30%;
    display: flex;
    align-items: center;
    justify-content: center;
    overflow: hidden;
    border-top-left-radius: 20px;
    border-top-right-radius: 20px;
    background-color: #FEE;
}

.srBanner img {
    width: 100%;
    height: auto;
}

.srInfo {
    width: 100%;
    height: 50%;
    display: flex;
    align-items: center;
    flex-direction: column;
    margin-top: -20px;
}


.srIcon {
    width: 40px;
    height: 40px;
    border-radius: 50%;
    display: flex;
    align-items: center;
    justify-content: center;
    overflow: hidden;
    background-color: #005792;
}

.srIcon img,.srIcon svg {
    width: 100%;
    height: 100%;
    color: #FF6B6B;
}

.srNameTitle {
    display: flex;
    flex-direction: column;
    align-items: center;
}

.srName {
    color: #005792;
    font-size: 15px;
    padding: 0px 3px;
    word-break: break-all;
    text-align: center;
}


.srButtons {
    width: 100%;
    height: 30%;
    display: flex;
    align-items: center;
    justify-content: space-around;
}

.srDetail,.srDeleteOrAdd {
    width: 30px;
    height: 30px;
    background-color: transparent;
    cursor: pointer;
    border: none;
}

.srDetail svg,.srDeleteOrAdd svg {
    width: 100%;
    height: 100%;
    color: #FF6B6B;
}

@media only screen and (max-width: 520px) {
    .srCon {
        width: 120px;
        height: 120px;
        border-radius: 20px;
    }
    .srIcon {
        width: 30px;
        height: 30px;
    }
    .srDetail,.srDeleteOrAdd {
        width: 25px;
        height: 25px;
    }
    .srName {
        font-size: 13px;
    }
}