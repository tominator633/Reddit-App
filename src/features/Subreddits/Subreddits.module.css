* {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    font-family: "Urbanist", sans-serif;
}

/* 
  .gb {
    border: 1px solid red;
}    
 */

/* NAV SUBREDDITS / TOPICS COMPONENT */
 .swiper {
    width: 100%;
    height: 100%;
    display: flex;
    align-items: center;
  justify-content: space-between;
}
.swiperSlide {
    display: flex;
    align-items: center;
    justify-content: space-around;
}
#topic {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    text-decoration: none;
}

#topicIcon {
    width: 40px;
    height: 40px;
    border-radius: 50%;
}
#topicIcon svg {
    width: 100%;
    height: 100%;
    color: #FF6B6B;
}
#topicIcon img {
    width: 100%;
    height: 100%;
    border-radius: 50%;
}
#topic h3 {
    color: #DAF6FB;
    font-size: 14px;
}
.activeNavLink {
    font-weight: 700;
}
.inactiveNavLink {
    font-weight: 100;
}


#btnPrev,
#btnNext {
    color: #DAF6FB;
    width: 25px;
    height: 25px;
    background-color: #FF6B6B;
    border: none;
    border-radius: 50%;
}

#btnPrev::after,
#btnNext::after {
    font-size: 12px;
    font-weight: 700;
}