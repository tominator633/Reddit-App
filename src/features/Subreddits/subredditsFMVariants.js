



export const submitBtnVar = {
    visible: {
        marginTop: "0px",
        opacity: 1,
        transition: {
            duration: 0.3
        }
    },
    hidden: {
        marginTop: "-10px",
        opacity: 0,
        transition: {
            duration: 0.3
        }
    },
    }