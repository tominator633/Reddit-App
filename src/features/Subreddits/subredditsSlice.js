import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
const baseUrl = "https://www.reddit.com";

/* export const loadSubredditData = createAsyncThunk(
    "subreddits/loadSubredditIcon",
    async (subreddit) => {
        const searchEndpoint = `/r/${subreddit}/about.json`;
        const response = await fetch(baseUrl + searchEndpoint);
        if (response.ok) {
            const jsonResponse = await response.json();
            const subredditData = jsonResponse.data;
            return {
                name: subredditData.display_name,
                id: subredditData.id,
                title: subredditData.title,
                url: subredditData.url,
                description: subredditData.public_description,
                subscribers: subredditData.subscribers,
                iconSrc: subredditData.icon_img,
            }
        }
    }
); */
export const loadInitialSwiperSubreddit = createAsyncThunk(
    "subreddits/loadInitialSwiperSubreddit",
    async (subreddit) => {
        const searchEndpoint = `/r/${subreddit}/about.json`;
        const response = await fetch(baseUrl + searchEndpoint);
        if (response.ok) {
            const jsonResponse = await response.json();
            const subredditData = jsonResponse.data;
            return {
                name: subredditData.display_name,
                id: subredditData.id,
                subscribers: subredditData.subscribers,
                url: subredditData.url,
                headerTitle: subredditData.header_title,
                iconImg: subredditData.icon_img,
                headerImg: subredditData.header_img,
                bannerImg: subredditData.banner_img,
                publicDescription: subredditData.public_description,
            }
        }
    }
);



export const subredditsSlice = createSlice({
    name: "subreddits",
    initialState: {
        swiperSubreddits: ["funny", "AskReddit", "gaming", "worldnews", "movies", "science"],
        isLoading: false,
        hasError: false
    },
    reducers: {
        addSubreddit: (state, action) => {
            state.swiperSubreddits.push(action.payload);
        },
        deleteSubreddit: (state, action) => {
            return state.swiperSubreddits.filter(item => item !== action.payload)
        },
    },
    extraReducers: {
        [loadInitialSwiperSubreddit.pending]: (state) => {
            state.isLoading = true;
            state.hasError = false;
        },
        [loadInitialSwiperSubreddit.rejected]: (state) => {
            state.isLoading = false;
            state.hasError = true;
        },
        [loadInitialSwiperSubreddit.fulfilled]: (state, action) => {
            state.isLoading = false;
            state.hasError = false;
           state.swiperSubreddits.forEach((subreddit, index) => {
            if (subreddit === action.payload.name) {
              state.swiperSubreddits[index] = action.payload;
            }
          });
          //forEach is mutable. Immutable version would include map method and returning a new array
          
        }
    }
});

export const selectSwiperSubreddits = (state) => state.subreddits.swiperSubreddits;
export const selectIsLoading = (state) => state.subreddits.isLoading;
export const selectHasError = (state) => state.subreddits.hasError;
export const {addSubreddit, deleteSubreddit} = subredditsSlice.actions;



export default subredditsSlice.reducer;