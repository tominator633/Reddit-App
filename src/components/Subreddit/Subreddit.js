import React from "react";
import styles from "./Subreddit.module.css";
import { Link } from "react-router-dom";
import { useDispatch } from 'react-redux';
import { addSubreddit, deleteSubreddit, setCurrentSubreddit} from "../../features/Subreddits/subredditsSlice";


export default function Subreddit ({content, isSwiperSubreddit}) {

    const dispatch = useDispatch();

    const handleAddSubredditClick = () => {
        dispatch(addSubreddit(content));
    }

    const handleDeleteSubredditClick = () => {
        dispatch(deleteSubreddit(content));
    }

    const handleSrDetailBtnClick = () => {
        dispatch(setCurrentSubreddit(content));
    }



    return (
        <section className={styles.srCon}
            aria-label={content.name}>
            <figure className={styles.srBanner}
                    role="presentation">
                {content.bannerImg &&
                <img src={content.bannerImg} 
                    alt={`${content.name} banner`} />
                }
                
            </figure>
            
            <div className={styles.srInfo}
                role="presentation">
                <figure className={styles.srIcon}
                        role="presentation">
                    {content.iconImg ? 
                    <img src={content.iconImg} 
                        alt={`${content.name} icon`}/>
                    : content.headerImg ?
                    <img src={content.headerImg} 
                        alt={`${content.name} icon`}/>
                    :
                    <svg aria-hidden="true" xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" viewBox="0 0 24 24"><path fill="currentColor" d="M10.75 13.04c0-.57-.47-1.04-1.04-1.04c-.57 0-1.04.47-1.04 1.04a1.04 1.04 0 1 0 2.08 0m3.34 2.37c-.45.45-1.41.61-2.09.61s-1.64-.16-2.09-.61a.26.26 0 0 0-.38 0a.26.26 0 0 0 0 .38c.71.71 2.07.77 2.47.77c.4 0 1.76-.06 2.47-.77a.26.26 0 0 0 0-.38c-.1-.1-.27-.1-.38 0m.2-3.41c-.57 0-1.04.47-1.04 1.04c0 .57.47 1.04 1.04 1.04s1.04-.47 1.04-1.04c0-.57-.46-1.04-1.04-1.04"/><path fill="currentColor" d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10s10-4.48 10-10S17.52 2 12 2m5.8 11.33c.02.14.03.29.03.44c0 2.24-2.61 4.06-5.83 4.06s-5.83-1.82-5.83-4.06c0-.15.01-.3.03-.44c-.51-.23-.86-.74-.86-1.33a1.455 1.455 0 0 1 2.47-1.05c1.01-.73 2.41-1.19 3.96-1.24l.74-3.49c.01-.07.05-.13.11-.16c.06-.04.13-.05.2-.04l2.42.52a1.04 1.04 0 1 1 .93 1.5c-.56 0-1.01-.44-1.04-.99l-2.17-.46l-.66 3.12c1.53.05 2.9.52 3.9 1.24a1.455 1.455 0 1 1 1.6 2.38"/></svg>
                    }
                </figure>
                <div className={styles.srNameTitle}
                        role="presentation">
                    <h3 className={styles.srName}>{content.name}</h3>
                </div>
                
            </div>
            <div className={styles.srButtons}
                    role="presentation">
                <Link to={`${content.id}`}
                        className={styles.srDetail}
                        onClick={handleSrDetailBtnClick}
                        aria-label="Open information window for this subreddit"
                        >
                    <svg aria-hidden="true" xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" viewBox="0 0 24 24"><path fill="currentColor" d="M11 17h2v-6h-2zm1-8q.425 0 .713-.288T13 8t-.288-.712T12 7t-.712.288T11 8t.288.713T12 9m0 13q-2.075 0-3.9-.788t-3.175-2.137T2.788 15.9T2 12t.788-3.9t2.137-3.175T8.1 2.788T12 2t3.9.788t3.175 2.137T21.213 8.1T22 12t-.788 3.9t-2.137 3.175t-3.175 2.138T12 22m0-2q3.35 0 5.675-2.325T20 12t-2.325-5.675T12 4T6.325 6.325T4 12t2.325 5.675T12 20m0-8"/></svg>
                </Link>
                <button className={styles.srDeleteOrAdd}
                        onClick={isSwiperSubreddit ? handleDeleteSubredditClick : handleAddSubredditClick}
                        aria-label={isSwiperSubreddit ? "Remove subreddit from your selection" : "Add subreddit to your selection"}>
                    {isSwiperSubreddit ? 
                    <svg aria-hidden="true" xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" viewBox="0 0 24 24"><path fill="currentColor" d="M7 21q-.825 0-1.412-.587T5 19V6H4V4h5V3h6v1h5v2h-1v13q0 .825-.587 1.413T17 21zm2-4h2V8H9zm4 0h2V8h-2z"/></svg>
                    :
                    <svg aria-hidden="true" xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" viewBox="0 0 48 48"><defs><mask id="ipSAddOne0"><g fill="none" stroke-linejoin="round" stroke-width="4"><path fill="#fff" stroke="#fff" d="M24 44c11.046 0 20-8.954 20-20S35.046 4 24 4S4 12.954 4 24s8.954 20 20 20Z"/><path stroke="#000" stroke-linecap="round" d="M24 16v16m-8-8h16"/></g></mask></defs><path fill="currentColor" d="M0 0h48v48H0z" mask="url(#ipSAddOne0)"/></svg>
                    }
                </button>
            </div>
        </section>
    )
}
