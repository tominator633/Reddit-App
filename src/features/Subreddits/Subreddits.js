import React, {useState, useRef} from "react";
import styles from "./Subreddits.module.css";
import { motion, AnimatePresence, LayoutGroup } from 'framer-motion';
import {submitBtnVar, mySubredditVar, searchedSubredditVar} from "./subredditsFMVariants";
import Subreddit from "../../components/Subreddit/Subreddit";
import Loading from "../../components/Loading/Loading";
import ErrorMessage from "../../components/ErrorMessage/ErrorMessage";
import { Outlet } from 'react-router-dom';
import {useSelector, useDispatch } from 'react-redux';
import {selectSwiperSubreddits, selectSearchedSubreddits, searchSubreddits, selectIsSearchSubredditsLoading, selectHasSearchSubredditsError} from "./subredditsSlice";

export default function Subreddits () {

    const dispatch = useDispatch();

    const swiperSubreddits = useSelector(selectSwiperSubreddits);
    const searchedSubreddits = useSelector(selectSearchedSubreddits);
    const isSearchSubredditsLoading = useSelector(selectIsSearchSubredditsLoading);
    const hasSearchSubredditsError = useSelector(selectHasSearchSubredditsError);

    const [searchInput, setSearchInput] = useState("");
    const searchInputRef = useRef(null);

    const sanitizeInput = (input) => {
        const tempElement = document.createElement('div');
        tempElement.textContent = input; // Sanitizes input
        return tempElement.innerHTML;
    };

    const handleSearchFieldChange = ({ target }) => {
        const regex = /^[A-Za-zÀ-ÖØ-öø-ÿ0-9\sěščřžůú]*$/;
        if (regex.test(target.value)) {
            setSearchInput(target.value); 
        } else {
            alert("Invalid input. Only alphanumeric characters, specific diacritics and spaces are allowed.");
        }
    };


    const handleSubmitSearchSubredditsBtnClick = () => {
        const sanitizedInput = sanitizeInput(searchInput);
        const encodedInput = encodeURIComponent(sanitizedInput);  // Encode input
        dispatch(searchSubreddits(encodedInput));
    };

    const handleKeyDown = (event) => {
        if (event.key === 'Enter' && searchInput) {
            event.preventDefault();
            const sanitizedInput = sanitizeInput(searchInput);
            const encodedInput = encodeURIComponent(sanitizedInput);
            dispatch(searchSubreddits(encodedInput));
        }
    };

    const handleErrorSearchSubmitReloadClick = () => {
        const sanitizedInput = sanitizeInput(searchInput);
        const encodedInput = encodeURIComponent(sanitizedInput);  // Encode the sanitized input
        dispatch(searchSubreddits(encodedInput));  // Dispatch the encoded input
    };
    return (
        <>
        <div className={`${styles.srManagerCon} ${styles.gb}`}>
            <section className={`${styles.mySubredditsCon} ${styles.gb}`}>
                <h2 className={`${styles.mySubredditsH2} ${styles.gb}`}>My Subreddits selection</h2>
                <div className={`${styles.mySubreddits} ${styles.gb}`}>
                <AnimatePresence> 
                    {swiperSubreddits.length > 0 ?
                    swiperSubreddits.map((subreddit) => {
                    return    (
                        <LayoutGroup key={subreddit.id}>
                            <motion.div variants={mySubredditVar}
                                        layout
                                        initial="hidden"
                                        animate="visible"
                                        exit="exit"
                                        transition={{ duration: 0.2 }}>
                                <Subreddit content={subreddit} 
                                            key={subreddit.id}
                                            isSwiperSubreddit={true}/>
                            </motion.div>
                        </LayoutGroup>
                        )
                        } 
                    )
                    :
                    <ErrorMessage message="You have no subreddits in your selection." />
                    }
                </AnimatePresence>
                </div>
            </section>
            <section className={`${styles.searchSubredditsCon} ${styles.gb}`}>
                <h2 className={`${styles.searchSubredditsH2} ${styles.gb}`}>Explore subreddits</h2>
                <search className={styles.searchSubredditsSection}
                        onKeyDown={handleKeyDown}>
                    <input className={styles.searchField} 
                            onChange={handleSearchFieldChange}
                            id="searchSubredditsField"
                            value={searchInput}
                            ref={searchInputRef}
                            placeholder="Search subreddits here"
                            maxLength="60"
                            pattern="[A-Za-z0-9\s]+"
                            title="Alphanumeric characters only"/>
                </search>
                <AnimatePresence>
                {searchInput &&
                <motion.button className={`${styles.submitSearchSubredditsBtn} ${styles.gb}`}
                        onClick={handleSubmitSearchSubredditsBtnClick}
                        
                        variants={submitBtnVar}
                        initial="hidden"
                        animate="visible"
                        exit="hidden"
                        
                        >
                    <svg role="presentation" xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" viewBox="0 0 24 24"><path fill="currentColor" d="M15.5 14h-.79l-.28-.27a6.5 6.5 0 0 0 1.48-5.34c-.47-2.78-2.79-5-5.59-5.34a6.505 6.505 0 0 0-7.27 7.27c.34 2.8 2.56 5.12 5.34 5.59a6.5 6.5 0 0 0 5.34-1.48l.27.28v.79l4.25 4.25c.41.41 1.08.41 1.49 0c.41-.41.41-1.08 0-1.49zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5S14 7.01 14 9.5S11.99 14 9.5 14"></path></svg>
                </motion.button>
                }
                </AnimatePresence>
                <div className={`${styles.searchedSubreddits} ${styles.gb}`}>
                <AnimatePresence> 
                    {isSearchSubredditsLoading ?
                    <Loading loadingText="Loading subreddits..."/>
                    : hasSearchSubredditsError ?
                    <ErrorMessage message="Request failed."
                                    onClick={handleErrorSearchSubmitReloadClick} />
                    : searchedSubreddits.length === 0 ?
                    <ErrorMessage message="No subreddits found" />
                    :
                    searchedSubreddits.map((subreddit) => {
                    return (
                        <LayoutGroup key={subreddit.id}>
                            <motion.div variants={searchedSubredditVar}
                                        layout
                                        initial="hidden"
                                        animate="visible"
                                        exit="exit"
                                        transition={{ duration: 0.2 }}>
                                <Subreddit content={subreddit} 
                                            key={subreddit.id}
                                            isSwiperSubreddit={false}/>
                            </motion.div>
                        </LayoutGroup>
                    )
                    })
                    }
                </AnimatePresence>
                </div>
                
                
            </section>
            
        </div>
        <Outlet/>
        </>
    )
};



/* the Outlet is just below the outermost element, in 
order for the backdrop filter to blur everything in behind */

