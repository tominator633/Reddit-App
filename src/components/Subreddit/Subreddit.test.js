import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { addSubreddit, deleteSubreddit, setCurrentSubreddit } from '../../features/Subreddits/subredditsSlice';
import Subreddit from './Subreddit';

const mockStore = configureStore([]);

describe('Subreddit component', () => {
    let store;
    let content;

    beforeEach(() => {
        store = mockStore({});
        store.dispatch = jest.fn();

        content = {
            id: 'test-subreddit',
            name: 'Test Subreddit',
            bannerImg: 'test-banner.png',
            iconImg: 'test-icon.png',
            headerImg: 'test-header.png'
        };
    });

    const renderComponent = (isSwiperSubreddit = false) => {
        render(
            <Provider store={store}>
                <MemoryRouter>
                    <Subreddit content={content} isSwiperSubreddit={isSwiperSubreddit} />
                </MemoryRouter>
            </Provider>
        );
    };

    it('renders subreddit name, banner, and icon if present', () => {
        renderComponent();

        expect(screen.getByRole('img', { name: 'Test Subreddit banner' })).toBeInTheDocument();
        expect(screen.getByRole('img', { name: 'Test Subreddit icon' })).toBeInTheDocument();
        expect(screen.getByText('Test Subreddit')).toBeInTheDocument();
    });

    it('displays the header image if the icon image is not available', () => {
        content.iconImg = null;
        renderComponent();

        expect(screen.getByRole('img', { name: 'Test Subreddit icon' })).toBeInTheDocument();
        expect(screen.queryByAltText('test-icon.png')).not.toBeInTheDocument();
    });

    it('renders a default icon if no icon or header image is provided', () => {
        content.iconImg = null;
        content.headerImg = null;
        renderComponent();

        expect(screen.getByRole('presentation').querySelector('svg')).toBeInTheDocument();
    });

    it('dispatches setCurrentSubreddit when detail link is clicked', () => {
        renderComponent();
        fireEvent.click(screen.getByRole('link', { name: /Open information window for this subreddit/i }));

        expect(store.dispatch).toHaveBeenCalledWith(setCurrentSubreddit(content));
    });

    it('dispatches addSubreddit when "Add subreddit" button is clicked', () => {
        renderComponent(false);
        fireEvent.click(screen.getByRole('button', { name: /Add subreddit to your selection/i }));

        expect(store.dispatch).toHaveBeenCalledWith(addSubreddit(content));
    });

    it('dispatches deleteSubreddit when "Remove subreddit" button is clicked if isSwiperSubreddit is true', () => {
        renderComponent(true);
        fireEvent.click(screen.getByRole('button', { name: /Remove subreddit from your selection/i }));

        expect(store.dispatch).toHaveBeenCalledWith(deleteSubreddit(content));
    });

    it('matches the snapshot for the component', () => {
        const { asFragment } = render(
            <Provider store={store}>
                <MemoryRouter>
                    <Subreddit content={content} isSwiperSubreddit={false} />
                </MemoryRouter>
            </Provider>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
