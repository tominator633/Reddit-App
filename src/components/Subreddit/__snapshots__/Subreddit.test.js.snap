// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subreddit component matches the snapshot for the component 1`] = `
<DocumentFragment>
  <section
    aria-label="Test Subreddit"
    class="srCon"
  >
    <figure
      class="srBanner"
      role="presentation"
    >
      <img
        alt="Test Subreddit banner"
        src="test-banner.png"
      />
    </figure>
    <div
      class="srInfo"
      role="presentation"
    >
      <figure
        class="srIcon"
        role="presentation"
      >
        <img
          alt="Test Subreddit icon"
          src="test-icon.png"
        />
      </figure>
      <div
        class="srNameTitle"
        role="presentation"
      >
        <h3
          class="srName"
        >
          Test Subreddit
        </h3>
      </div>
    </div>
    <div
      class="srButtons"
      role="presentation"
    >
      <a
        aria-label="Open information window for this subreddit"
        class="srDetail"
        href="/test-subreddit"
      >
        <svg
          aria-hidden="true"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11 17h2v-6h-2zm1-8q.425 0 .713-.288T13 8t-.288-.712T12 7t-.712.288T11 8t.288.713T12 9m0 13q-2.075 0-3.9-.788t-3.175-2.137T2.788 15.9T2 12t.788-3.9t2.137-3.175T8.1 2.788T12 2t3.9.788t3.175 2.137T21.213 8.1T22 12t-.788 3.9t-2.137 3.175t-3.175 2.138T12 22m0-2q3.35 0 5.675-2.325T20 12t-2.325-5.675T12 4T6.325 6.325T4 12t2.325 5.675T12 20m0-8"
            fill="currentColor"
          />
        </svg>
      </a>
      <button
        aria-label="Add subreddit to your selection"
        class="srDeleteOrAdd"
      >
        <svg
          aria-hidden="true"
          height="1em"
          viewBox="0 0 48 48"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs>
            <mask
              id="ipSAddOne0"
            >
              <g
                fill="none"
                stroke-linejoin="round"
                stroke-width="4"
              >
                <path
                  d="M24 44c11.046 0 20-8.954 20-20S35.046 4 24 4S4 12.954 4 24s8.954 20 20 20Z"
                  fill="#fff"
                  stroke="#fff"
                />
                <path
                  d="M24 16v16m-8-8h16"
                  stroke="#000"
                  stroke-linecap="round"
                />
              </g>
            </mask>
          </defs>
          <path
            d="M0 0h48v48H0z"
            fill="currentColor"
            mask="url(#ipSAddOne0)"
          />
        </svg>
      </button>
    </div>
  </section>
</DocumentFragment>
`;
